specs/001-design-and-implement/tasks.md

```
@@ -0,0 +1,203 @@
+- [] T014 [P] Snapshot test Linux assembly output in tests/test_linux_assembly_snapshots.rs
+- [] T015 [P] Snapshot test Windows assembly output in tests/test_windows_assembly_snapshots.rs
+- [] T016 [P] Snapshot test macOS assembly output in tests/test_macos_assembly_snapshots.rs
+
+## Phase 3.3: Core Type Definitions (ONLY after tests are failing) [P]
+**All enum and struct definitions can run in parallel - different files**
+
+### Register and Instruction Types [P]
+- [] T017 [P] GPRegister enum with Display trait in src/asm/register.rs
+- [] T018 [P] MMRegister enum with conversion methods in src/asm/register.rs
+- [] T019 [P] Register unified enum in src/asm/register.rs
+- [] T020 [P] 86Instruction enum with all instruction variants in src/asm/instruction.rs
```

markdownlint-cli2 (0.18.1)

Headings should be surrounded by blank lines
Expected: 1; Actual: 0; Below

(MD022, blanks-around-headings)